src/org/traccar/database/DeviceManager.java

```diff
@@ -262,59 +262,44 @@ public Position getLastPosition(long deviceId) {
 
     public boolean lookupAttributeBoolean(
             long deviceId, String attributeName, boolean defaultValue, boolean lookupConfig) {
-        String result = lookupAttribute(deviceId, attributeName, lookupConfig);
-        if (result != null) {
-            return Boolean.parseBoolean(result);
-        }
-        return defaultValue;
+        Object result = lookupAttribute(deviceId, attributeName, lookupConfig);
+        return result != null ? Boolean.parseBoolean(result.toString()) : defaultValue;
```

Comment:
I don't think it's the right way. It's better to check object type, and only parse if it's a string. Applies to all other methods.

Reply:
Something like this?
````java
    public boolean lookupAttributeBoolean(
            long deviceId, String attributeName, boolean defaultValue, boolean lookupConfig) {
        Object result = lookupAttribute(deviceId, attributeName, lookupConfig);
        if (result != null) {
            if (result instanceof String) {
                return Boolean.parseBoolean((String) result);
            } else {
                return (Boolean) result;
            }
        } else {
            return defaultValue;
        }
    }

````

Reply:
Yes, maybe something like this to make it shorter:
```java
return result instanceof String ? Boolean.parseBoolean((String) result) : (Boolean) result;
```